File: 3477 Fruits Into Baskets II.cpp
class Solution {
public:
    int numOfUnplacedFruits(vector<int>& fruits, vector<int>& baskets) {
        int unplaced = fruits.size();
        list<int> baskets_list(baskets.begin(), baskets.end());
        for (int fruit : fruits) {
            for (auto it = baskets_list.begin(); it != baskets_list.end(); ++it) {
                if (fruit <= *it) {
                    baskets_list.erase(it);
                    unplaced--;
                    break;
                }
            }
        }
        return unplaced;
    }
};

File: 3477 Fruits Into Baskets II.rs
impl Solution {
    pub fn num_of_unplaced_fruits(fruits: Vec<i32>, baskets: Vec<i32>) -> i32 {
        let mut baskets_copy = baskets.clone();
        let mut count = fruits.len() as i32;
        for fruit in fruits {
            for basket in &mut baskets_copy {
                if *basket >= fruit {
                    count -= 1;
                    *basket = 0; // Mark this basket as used
                    break;
                }
            }
        }

        count
    }
}